### PR TITLE
dropdown menus for sidebar buttons; closes #1010

### LINKED ISSUE #1010: Dropdown menu for Quests and Badges
Carve off the right side of the Quests and Badges buttons in the left menu, and place drop down menus there:
image

Content for now:

Quests drop down menu items:
- In Progress (students)
- Completed (students)
- Drafts (contextual, if they have a drafts tab - TAs and staff)
- Campaigns (staff)
- Common quest info (staff)

Badges drop down menu items:
- Badge Types (staff)

### PR DESCRIPTION
**Pull request by Dylan-RG-Patterson**
sidebar buttons now have dropdown menus to make larger portions of the site more accessible.
image
dropdown menus vary depending on whether or not the user is a staff member or not.
image
image

### CODE REVIEW

File: src/quest_manager/templates/quest_manager/category_list.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% extends "quest_manager/base.html" %}
+{% extends "courses/base.html" %}
```
Comment (tylerecouture) [inline src/quest_manager/templates/quest_manager/category_list.html:1]: Why this change? Campaign/categories are in the quest_manager app (models/views, etc)

File: src/templates/sidebar.html [hunk 2]
```diff
@@ -17,34 +17,62 @@ <h3 class="hidden-xs">
...
-    {% if request.user.is_staff %}
+
+      <div id="dropdown-button-wrapper"> <!-- isolates dropdown button highlight -->
+        <a id="quests-menu-dropdown" 
+          class="list-group-item right-side dropdown-toggle text-center" 
+          data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
+          <i class="fa fa-ellipsis-v"></i></a>
+        <ul class="dropdown-menu dropdown-menu-right">
+          {% if request.user.is_staff %}
+            <li><a href="{% url 'quests:categories' %}"><i class="fa fa-compass"></i>&nbsp; Campaigns</a></li>
```
Comment (tylerecouture) [inline src/templates/sidebar.html:32]: Add `fa-fw` to the icons to get text to line up vertically after an icon (font awesome fixed-width, makes all icon elements the same width by adjusting padding margins)

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Awesome! This looks great! I noted a couple minor things.

File: src/quest_manager/templates/quest_manager/category_list.html (hunk 1 above)
Comment (Dylan-RG-Patterson) [reply src/quest_manager/templates/quest_manager/category_list.html]: the change was originally intended to keep consistent with the top right icon on the campaigns page and the icon used to represent it in the admin menu, but there is an inconsistency with site logic. change reverted.
image

File: src/templates/sidebar.html (hunk 2 above)
Comment (Dylan-RG-Patterson) [reply src/templates/sidebar.html]: fa-fw tag added to fontawesome icons.

**Review by tylerecouture — APPROVED** [review]

**Review by tylerecouture — APPROVED** [review]
